### ISSUE TITLE
Add support for mirroring auto routines

### ISSUE THREAD

**Issue by WesleyAC** [donald-pinckney/lemonscript#27]
Since the field is mirrored this year, we will need to mirror auto routines.

It would be good to automatically do this in lemonscript. This will likely take some thinking about, and potentially some work in transpiler as well.

Would be good to talk about in person next time you're at a meeting.

**Comment by donald-pinckney** [comment]
My mirroring, do you mean basically doing:
PointTurn: 1.5 * rad
vs
PointTurn: -1.5 * rad
???

**Comment by donald-pinckney** [comment]
If so, #10 is probably a good way to do this, possibly #11 would help.

**Comment by WesleyAC** [comment]
Yes, that's what I mean.

I'm not really sure how #10 and #11 help a ton. I guess it depends on how you implement it. I was imagining something where functions would define certain args as args that should be flipped, and there's a mirror setting in the runner.

I suppose that you could set a variable to -1 or 1 for what side it's on, and then multiply everything by that number, but that's kinda a hack.

**Comment by donald-pinckney** [comment]
Yeah, I was thinking that with #10 you could do something like:
```
DEF flipped = FlippedConstant() // Either 1 or -1
PointTurn: 1.5 * flipped * rad
```
I kinda want to avoid making lemonscript changes that are highly year specific...

**Comment by WesleyAC** [comment]
Yeah, I know what you mean about not wanting to get year specific.

I'm just worried abut the verbosity of always having to add ` * flipped`. And especially worried about forgetting to add it at all.

Maybe the best way is to just do it in the actual lemonscript function definitions...

**Comment by donald-pinckney** [comment]
Yeah, I agree about the verbosity.  The lemonscript function definitions are certainly a decent way to do it. I'll let you know if I think of any other ideas though.

**Comment by WesleyAC** [comment]
I could also do a jank thing where I have:

auto_routine.auto:
```
DoSomeStuff
...
DoMoreStuff
```

auto_routine_flipped.auto:
```
SET flipped = -1
IMPORT auto_routine.auto
```

Or do an equivalent to that where I use a C++ function to set if it's flipped.

**Comment by donald-pinckney** [comment]
It's slightly less jank if you use `RUN auto_routine.auto`.

I'm not sure what this accomplishes, compared to just doing basically the same thing internally in the functions.
